diff --git a/pEpForiOS/Info.plist b/pEpForiOS/Info.plist
index bd7d70308..1d4300aed 100644
--- a/pEpForiOS/Info.plist
+++ b/pEpForiOS/Info.plist
@@ -4,7 +4,7 @@
 <dict>
 	<key>BGTaskSchedulerPermittedIdentifiers</key>
 	<array>
-		<string>security.pep.pep4ios.backgroundtaskschedulerid.backgroundfetch</string>
+		<string>software.pEp.mail.background-refresh</string>
 	</array>
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en_US</string>
diff --git a/pEpForiOS/Util/Constants.swift b/pEpForiOS/Util/Constants.swift
index a76813d30..a81df44b5 100644
--- a/pEpForiOS/Util/Constants.swift
+++ b/pEpForiOS/Util/Constants.swift
@@ -49,5 +49,5 @@ extension Constants {
 
 extension Constants {
     static let appRefreshTaskBackgroundtaskBackgroundfetchSchedulerid =
-        "security.pep.pep4ios.backgroundtaskschedulerid.backgroundfetch"
+        "software.pEp.mail.background-refresh"
 }
